internal/utils/errors.go

```diff
@@ -17,6 +17,11 @@ var (
 	ErrIOTimeout             = errors.New("i/o timeout")
 )
 
+var (
+	// ErrNotImplemented is returned when a function is not implemented.
+	ErrNotImplemented = errors.New("not implemented")
```

Comment:
I think if you hit something that is not implemented you are allowed to just `panic`.